sim-lib/src/lib.rs

```diff
@@ -349,6 +349,19 @@ pub struct WriteResults {
     pub batch_size: u32,
 }
 
+/// PayGenDisplay is a marker trait to allow creation of trait objects for printable payment generators.
+pub trait PayGenDisplay: PaymentGenerator + Display {}
```

Comment:
Since all our payment generators impl display, I think we can just define `pub trait PaymentGenerator: Display {}` and avoid this additional marker trait

Reply:
TIL supertraits exist :')

Reply:
:)